reflect/aipreflect/validateresourcereferences.go

```diff
@@ -0,0 +1,75 @@
+package aipreflect
+
+import (
+	"fmt"
+	"strings"
+
+	"go.einride.tech/aip/resourcename"
+	"google.golang.org/genproto/googleapis/api/annotations"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/reflect/protopath"
+	"google.golang.org/protobuf/reflect/protorange"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+)
+
+// ValidateResourceReferences validates the resource reference fields in a message.
+func ValidateResourceReferences(message proto.Message) error {
+	return protorange.Range(message.ProtoReflect(), func(values protopath.Values) error {
+		curr := values.Index(-1)
+		var field protoreflect.FieldDescriptor
+		var fieldValue string
+		switch curr.Step.Kind() {
+		case protopath.FieldAccessStep:
+			field = curr.Step.FieldDescriptor()
+			if field.Kind() != protoreflect.StringKind {
+				return nil
+			}
+			if field.Cardinality() == protoreflect.Repeated {
+				return nil
+			}
+			fieldValue = curr.Value.String()
+		case protopath.ListIndexStep:
+			prev := values.Index(-2)
+			field = prev.Step.FieldDescriptor()
+			if field.Kind() != protoreflect.StringKind {
+				return nil
+			}
+			fieldValue = curr.Value.String()
+		default:
+			return nil
+		}
+		resourceReferenceAnnotation := proto.GetExtension(
+			field.Options(), annotations.E_ResourceReference,
+		).(*annotations.ResourceReference)
+		if resourceReferenceAnnotation == nil {
+			return nil
+		}
+		var errValidate error
```

Comment:
use `status.Details` to validate everything?

Reply:
Instead of only returning the first error.

Reply:
I'm working on this, but I think we also want to be able to combine the field level validation with other validators - so let's take that as a separate step?